src/components/CreateCourse/CreateCourse.tsx

```diff
@@ -5,97 +5,92 @@ import { v4 as uuidv4 } from 'uuid';
 import AddTitle from './components/AddTitle/AddTitle';
 import AddDescriptions from './components/AddDescriptions/AddDescriptions';
 import AddDetails from './components/AddDetails/AddDetails';
-import Button from '../../common/Button/Button';
+import Button from 'common/Button/Button';
+import Duration from './components/AddDetails/components/Duration/Duration';
 
-import { VALID } from '../../constants';
-import * as fn from '../../helpers/index';
-import { useAuthorsReducer, useDataValidation } from '../../hooks/index';
-import { TypeCreateCourse, TypeAddDetails } from '../../types/types';
+import { ERROR, REGEX } from '../../constants';
+import * as fn from 'helpers/index';
+import { useDataValidation } from 'hooks';
+import { CourseID } from 'store/courses/coursesTypes';
+import { getAuthors, useAppDispatch, useAppSelector } from 'store/storeTypes';
+import { createCourse } from 'store/courses/coursesCreators';
+import { cleareList } from 'store/authors/authorsCreators';
 
 import styles from './CreateCourse.module.css';
 
-const CreateCourse: React.FC<TypeCreateCourse> = ({ ...props }) => {
+const CreateCourse = () => {
 	const navigate = useNavigate();
-	const [inputs, setInputs] = useState({ title: '', descriptions: '' });
+	const dispatch = useAppDispatch();
+	const authorsList = useAppSelector(getAuthors);
 
-	const [handleDuration, duration] = useDataValidation(
-		VALID.duration.messageError,
-		VALID.duration.regex
-	);
-	const [handleCheckName, checkName, setCheckName] = useDataValidation(
-		VALID.name.messageError,
-		VALID.name.regex
-	);
+	const duration = useDataValidation(ERROR.msg_3, REGEX.duration);
 
-	const [state, author, setAuthorName, createAuthor, addAuthor, deleteAuthor] =
-		useAuthorsReducer({
-			authors: props.authorsList,
-			setAuthors: props.setAuthorsList,
-		});
+	const [newCourse, setNewCourse] = useState<CourseID>({
+		id: uuidv4(),
+		title: '',
+		description: '',
+		creationDate: fn.dateGenerator(new Date()),
+		duration: 0,
+		authors: [],
+	});
```

Comment:
it is better to not use complicated states like this and just divide them in separate states

Reply:
I will keep that in mind.